### ISSUE TITLE
Firebase messaging onResume and onMessage broken on iOS since Flutter 1.7.4

### ISSUE THREAD

**Issue by JaspervanRiet** [flutter/flutter#36201]
On Flutter's dev channel (and all other channels), Firebase Messaging's onResume and onMessage callbacks are broken on iOS as of 1.7.4. We have done extensive testing, and it works fine with v1.6.3, and even v1.6.7. onLaunch works fine, and all callbacks work fine on Android. The callbacks are not called at all, and the following error can be found in the iOS logs:

```
SpringBoard(UserNotificationsServer)[52] <Notice>: [com.ourapp.someapp] Application foreground
launch action for notification response action com.apple.UNNotificationDefaultActionIdentifier
recieved action response <BSActionResponse: 0x2839cde00; error:
<NSError: 0x2834c37e0; domain: BSActionErrorDomain; code: 4> {
    description = "The operation couldn\M-b\M^@\M^Yt be completed. (BSActionErrorDomain error 4.)";
}>
```
Every now and then it shows `error 1` instead of `error 4`. This error is terribly documented online (not at all), but I recognise it from when we had our iOS app configurated wrong for push notifications. Our configuration is fine now. We build via a build server, so we still have all the logs. We built a version on June 20st (from now on referred to as A), and a version on June 21st (B).  A was v1.6.3, B was v1.7.4. We've since tested v1.6.7, and that version is fine too. Version A calls the callbacks and works fine. Version B does not. The issue is not related to our code since on our current HEAD, the callbacks still only work < v1.7.4.

Now, we have had to revert back to Flutter v1.6.7, meaning we no longer have access to app bundles and TransferableTypedData, which is, of course, a major blocker for our work.

Flutter version for A:
```
Flutter 1.6.3 • channel beta • https://github.com/flutter/flutter.git
Framework • revision bc7bc94083 (4 weeks ago) • 2019-05-23 10:29:07 -0700
Engine • revision 8dc3a4cde2
Tools • Dart 2.3.2 (build 2.3.2-dev.0.0 e3edfd36b2)
```

Flutter version for B:
```
Flutter 1.7.4 • channel dev • https://github.com/flutter/flutter.git
Framework • revision dfecafa4ab (7 days ago) • 2019-06-14 09:46:23 -0700
Engine • revision 2589785b5c
Tools • Dart 2.4.0
```

Firebase messaging version: 5.1.0, but applies to every version since at least 5.0.1 (that's as far as we went)

**Comment by VicM** [comment]
I am experiencing the same, I upgrade my project with flutter 1.5.8 pre-3 to v1.7.8+hotfix.2 and notifications stopped working on iOS. On android work well. Still waiting for any possible solution or workaround, any news?

**Comment by r0ck3r** [comment]
Same for me

**Comment by VicM** [comment]
I found a workaround, I will describe if that could help you, not sure if is 100% related to this issue. With the created XCode project make sure to check the "Automatically manage signing" option. I had that unchecked and had a developer provisoning profile manually assigned (with push notifications capabilities enabled) but with that setting the notifications didn't arrive, when using the automatic signing the push notifications started to arrive.

**Comment by JaspervanRiet** [comment]
@VicM We have always used that setting, so for this is not the solution I'm afraid.

**Comment by JaspervanRiet** [comment]
We are surprised to see there has not been a response to this issue yet, nor even a label. I believe I should mention @collinjackson. We've made even more attempts to resolve this issue but still no success.

**Comment by ThinkDigitalSoftware** [comment]
I'm having this issue as well.
I'm also getting no notification sound and all my sounds and notifications are enabled for this app and device and IosNotificationSettings()
> Flutter 1.8.3 • channel beta • https://github.com/flutter/flutter.git
> Framework • revision e4ebcdf6f4 (5 weeks ago) • 2019-07-27 11:48:24 -0700
> Engine • revision 38ac5f30a7
> Tools • Dart 2.5.0 (build 2.5.0-dev.1.0 0ca1582afd)

**Comment by Rekubot** [comment]
I am also experiencing this issue, which will impact the release of our new app.

**Comment by Rekubot** [comment]
Disregard, I've fixed the issue I was having. I'd overridden the didRegisterForRemoteNotificationsWithDeviceToken and didReceiveRemoteNotification methods in AppDelegate without calling the super methods, so the Firebase callbacks were not being triggered. It now works on the latest stable version, v1.7.8+hotfix.4.

**Comment by JaspervanRiet** [comment]
I would urge people posting in this issue to first check if reverting to v1.6.7 works before posting, since this issue is specifically about the upgrade to v1.7.4 breaking the notifications. We do not have bundle ID issues or any other configuration issues, as shown by the fact that everything works fine. This issue is specifically about the upgrade breaking the plugin.

**Comment by ThinkDigitalSoftware** [comment]
> I would urge people posting in this issue to first check if reverting to v1.6.7 works before posting, since this issue is specifically about the upgrade to v1.7.4 breaking the notifications. We do not have bundle ID issues or any other configuration issues, as shown by the fact that everything works fine on v1.6.7. This issue is specifically about the upgrade breaking the plugin.

OK. I'll do that now and report back

**Comment by ThinkDigitalSoftware** [comment]
Reporting back. Downgrading did not change anything for me.  Found a comment in another issue that may be pointing to my issue. So, @JaspervanRiet, you are correct about this not being my issue. Here's a link to the post for others having this issue that's unrelated to the upgrade.
https://github.com/flutter/flutter/issues/35375#issuecomment-527649127

**Comment by collinjackson** [comment]
Messaging plugin is going to be rewritten by @bparrishMines once https://github.com/flutter/engine/pull/9864 lands, so we'll want to make sure that this issue is addressed as part of that change.

**Comment by JaspervanRiet** [comment]
This issue is now starting to impact our release date. To reiterate: to this day, we're having to compile our iOS app using v1.6.7. Notifications still work fine. We do not use the local_notifications plugin. Flutter upgrading to any version since v1.6.7. breaks the notifications. Notifications are still received in iOS, but tapping them only works if the app is not in the background. If it is, the app is opened, and then nothing happens. An error appears in the iOS logs (not the Flutter logs) with no real description. When the app is open, onMessage is never fired. 

As a result, we do not have access to TransferableTypedData (vital to our app), iOS 13 fixes (obviously vital), and all the other stuff that's been added since v1.6.7.

**Comment by kroikie** [comment]
@JaspervanRiet

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/1056. Any further collaboration will be done there.

**Comment by robertohuertasm** [comment]
@Rekubot can you provide more details about your solution?
